### PR TITLE
GH-2921: Remove Jackson JDK8 module

### LINKED ISSUE #2921: Remove Jackson JDK8 module
**Reporter**: [Fokko Driesprong](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fokko) / @Fokko
**Assignee**: [Fokko Driesprong](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=fokko) / @Fokko

<sub>**Note**: *This issue was originally created as [PARQUET-2501](https://issues.apache.org/jira/browse/PARQUET-2501). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

### PR DESCRIPTION
**Pull request by Fokko**
Not needed anymore :)

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references
  them in the PR title. For example, "PARQUET-1234: My Parquet PR"
    - https://issues.apache.org/jira/browse/PARQUET-XXX
    - In case you are adding a dependency, check if the license complies with
      the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines
  from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    1. Subject is limited to 50 characters (not including Jira issue reference)
    1. Subject does not end with a period
    1. Subject uses the imperative mood ("add", not "adding")
    1. Body wraps at 72 characters
    1. Body explains "what" and "why", not "how"

### Style
- [ ] My contribution adheres to the code style guidelines and Spotless passes.
    - To apply the necessary changes, run `mvn spotless:apply -Pvector-plugins`

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain Javadoc that explain what it does

Closes #2921

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -69,10 +69,9 @@
     <jackson.package>com.fasterxml.jackson</jackson.package>
     <jackson.version>2.17.1</jackson.version>
     <jackson-databind.version>2.17.1</jackson-databind.version>
-    <jackson-modules-java8.version>2.17.1</jackson-modules-java8.version>
     <japicmp.version>0.21.0</japicmp.version>
     <javax.annotation.version>1.3.2</javax.annotation.version>
-    <spotless.version>2.30.0</spotless.version>
+    <spotless.version>2.43.0</spotless.version>
```
Comment (nastra) [inline pom.xml:74]: seems like an unrelated change

**Review by nastra — APPROVED** [review]
LGTM once the spotless version isn't bumped in the same diff ;)

File: pom.xml (hunk 1 above)
Comment (Fokko) [reply pom.xml:74]: Yes, thanks. I wanted to split this out into a separate PR.
```suggestion
    <spotless.version>2.30.0</spotless.version>
```

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/ParquetMetadata.java [hunk 2]
```diff
@@ -20,29 +20,19 @@
...
- * Meta Data block stored in the footer of the file
+ * Metadata block stored in the footer of the file
  * contains file level (Codec, Schema, ...) and block level (location, columns, record count, ...) meta data
  */
 public class ParquetMetadata {
 
   private static final ObjectMapper objectMapper = new ObjectMapper();
 
-  static {
-    // Enable FAIL_ON_EMPTY_BEANS on objectmapper. Without this feature parquet-casdacing tests fail,
```
Comment (Fokko) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/ParquetMetadata.java:39]: Cascading is long gone :)

**Review by wgtmac — APPROVED** [review]
